acados/ocp_nlp/ocp_nlp_common.c

```diff
@@ -313,11 +318,14 @@ void ocp_nlp_dims_initialize(void *config_, int *nx, int *nu, int *ny, int *nbx,
     {
         dims->qp_solver->nx[ii] = nx[ii];
         dims->qp_solver->nu[ii] = nu[ii];
+        dims->qp_solver->nb[ii] = nbx[ii] + nbu[ii];
         dims->qp_solver->nbx[ii] = nbx[ii];
         dims->qp_solver->nbu[ii] = nbu[ii];
-        dims->qp_solver->nb[ii] = nbx[ii] + nbu[ii];
         dims->qp_solver->ng[ii] = ng[ii] + nh[ii];
         dims->qp_solver->ns[ii] = ns[ii];
+        dims->qp_solver->nsbx[ii] = 0;
+        dims->qp_solver->nsbu[ii] = 0;
+        dims->qp_solver->nsg[ii] = 0;
```

Comment:
why are those set to zero since we support soft constraints in the NLP? Isn't this inconsistent with line:

dims->qp_solver->ns[ii] = ns[ii]; 

?

Reply:
HPIPM actually does not use nsbx nsbu nsg but only ns, the others are there only for Branimir's work on making the other QP solver handle soft constraints: there it is needed to know their type

Reply:
ok I am adding a todo for now